### DISCUSSION TITLE
Magic byte is not 0xE9 upgrade 7.2.0 to 8.5.1

### DISCUSSION THREAD

**Discussion by I-DontLike** [arendst/Tasmota#17469] [Q&A]
![Immagine](https://user-images.githubusercontent.com/110459287/208947495-e9f8ad9e-13a2-4865-8ff4-70533db26186.jpg)
Hi,
i have a sonoff basic with tasmota 7.2.0 firmware, but when i try upgrate to 8.5.1 version, i have Magic byte is not 0xE9 error. 
i tried with minimal version but i see the same error.

**Comment by sfromis** [comment]
When upgrading from that old a version, you should not use a compressed `.gz` binary. If that's not the issue, post precise details.

Be careful, and follow the steps in https://tasmota.github.io/docs/Upgrading/

**Comment by I-DontLike** [comment]
Ok, if i use the bin file see Upload buffer miscompare, but in the firmware page the minimal version.bin  is not present

**Comment by arendst** [comment]
You're supposed to upgrade using the OTA server.

The easiest way of doing this is via the GUI (or serial) console.
1) So open the GUI console and change your otaurl with command ``otaurl http://ota.tasmota.com/tasmota/release-8.5.1/tasmota.bin``
2) Then type ``upload 1`` and wait until the system restarts several times.
3) If you see you're no longer running the minimal version continue with the next release in the upgrade path. Change otaurl with command ``otaurl http://ota.tasmota.com/tasmota/release-9.1.0/tasmota.bin.gz``
4) type ``upload 1`` again
5) by now you should know the drill

**Reply by I-DontLike** [reply]
i have correctly upgraded to version 9.1, but after upgrade to 12.3.1 i can't connect to the gui and the sonoff connect and disconnect to the router wifi. I see the device name in the wifi network but i can't connect to it. The green led blinks
I have tried a reset, i see tasmota-xxx but i can't connect

**Reply by Jason2866** [reply]
Could be it is stuck at minimal firmware. Most probably only a serial flash is the way to recover the device.

**Reply by I-DontLike** [reply]
I have flashed the device, but when i try upgrade to 12.3.1 version, i have again the same problem.
With 12.2.0 firmware work fine.

i think the last firmware has a bug
